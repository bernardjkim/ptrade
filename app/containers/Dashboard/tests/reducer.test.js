import { fromJS } from 'immutable';
import dashboardReducer from '../reducer';
import {
  changeSearch,
  changeTimeFrame,
  loadStockData,
  stockDataLoaded,
  stockDataLoadingError,
} from '../actions';

describe('dashboardReducer', () => {
  let state;
  beforeEach(() => {
    state = fromJS({
      search: false,
      loading: false,
      error: false,
      currentSymbol: false,
      stockData: false,
    });
  });
  it('returns the initial state', () => {
    expect(dashboardReducer(undefined, {})).toMatchSnapshot();
  });

  it('handles the changeSearch action', () => {
    expect(dashboardReducer(state, changeSearch('AAPL'))).toMatchSnapshot();
  });

  it('handles the changeTimeFrame action', () => {
    expect(dashboardReducer(state, changeTimeFrame(1))).toMatchSnapshot();
  });

  it('should handle the loadStockData action correctly', () => {
    expect(dashboardReducer(state, loadStockData())).toMatchSnapshot();
  });

  it('should handle the stockDataLoaded action correctly', () => {
    const fixture = [
      {
        name: 'My Repo',
      },
    ];
    const currentSymbol = 'AAPL';

    expect(
      dashboardReducer(state, stockDataLoaded(fixture, currentSymbol)),
    ).toMatchSnapshot();
  });

  it('should handle the stockDataLoadingError action correctly', () => {
    const fixture = {
      msg: 'Not found',
    };
    expect(
      dashboardReducer(state, stockDataLoadingError(fixture)),
    ).toMatchSnapshot();
  });
});
