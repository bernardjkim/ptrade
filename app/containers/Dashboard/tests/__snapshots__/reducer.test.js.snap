// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dashboardReducer handles the changeSearch action 1`] = `
Immutable.Map {
  "search": "AAPL",
  "symbol": false,
  "loading": false,
  "error": false,
  "chart": false,
  "quote": false,
}
`;

exports[`dashboardReducer handles the changeTimeFrame action 1`] = `
Immutable.Map {
  "search": false,
  "symbol": false,
  "loading": false,
  "error": false,
  "chart": false,
  "quote": false,
  "timeFrame": 1,
}
`;

exports[`dashboardReducer handles the selectSymbol action 1`] = `
Immutable.Map {
  "search": false,
  "symbol": "AAPL",
  "loading": false,
  "error": false,
  "chart": false,
  "quote": false,
}
`;

exports[`dashboardReducer returns the initial state 1`] = `
Immutable.Map {
  "search": false,
  "symbol": false,
  "timeFrame": 0,
  "loading": false,
  "error": false,
  "chart": false,
  "quote": false,
}
`;

exports[`dashboardReducer should handle the loadChart action correctly 1`] = `
Immutable.Map {
  "search": false,
  "symbol": false,
  "loading": true,
  "error": false,
  "chart": false,
  "quote": false,
}
`;

exports[`dashboardReducer should handle the loadQuote action correctly 1`] = `
Immutable.Map {
  "search": false,
  "symbol": false,
  "loading": true,
  "error": false,
  "chart": false,
  "quote": false,
}
`;
